fundingmanager.go

```diff
@@ -1264,31 +1202,36 @@ func (f *fundingManager) handleFundingCreated(fmsg *fundingCreatedMsg) {
 		go f.waitForFundingWithTimeout(completeChan, confChan,
 			timeoutChan)
 
+		var shortChanID *lnwire.ShortChannelID
+		var ok bool
 		select {
 		case <-timeoutChan:
 			// We did not see the funding confirmation before
 			// timeout, so we forget the channel.
 			deleteFromDatabase()
+			return
 		case <-f.quit:
 			// The fundingManager is shutting down, will resume
 			// wait for funding transaction on startup.
-		case shortChanID, ok := <-confChan:
+			return
+		case shortChanID, ok = <-confChan:
 			if !ok {
 				fndgLog.Errorf("waiting for funding confirmation" +
 					" failed")
 				return
 			}
+			// Fallthrough.
+		}
 
-			f.deleteReservationCtx(peerKey, fmsg.msg.PendingChannelID)
+		// Success, funding transaction was confirmed.
+		f.deleteReservationCtx(peerKey, fmsg.msg.PendingChannelID)
 
-			// Success, funding transaction was confirmed.
-			err := f.handleFundingConfirmation(completeChan,
-				shortChanID)
-			if err != nil {
-				fndgLog.Errorf("failed to handle funding"+
-					"confirmation: %v", err)
-				return
-			}
+		err := f.handleFundingConfirmation(completeChan,
```

Comment:
`handleFundingConfirmation` is altered to also announce first after 6 confs. I should really make the state machine to centralize this logic...

Reply:
Yeah the state machine could really use a refactoring at this point. It sprawls far too much with this additional set of changes.

Reply:
I think I'll push the state machine refactoring to a new PR once this one is merged. This _should_ work (even though it's messy), and that PR should not change any functionality. Adding the state machine cleanup to this PR looks like it might become a bit messy.